source/mesh_refinement/isosurfaces.cc

```diff
@@ -0,0 +1,401 @@
+                            }
+                        }
+
+                      if ( isosurface.are_all_values_in_range(values))
+                        {
+                          // If the current refinement level is smaller than or equal to the minimum
+                          // refinement level, any coarsening flag should be cleared.
+                          if (cell->level() <= isosurface.min_refinement)
+                            {
+                              clear_coarsen = true;
+                            }
+
+                          // If the current refinement level is smaller than the minimum
+                          // refinement level, a refinement flag should be placed.
+                          if (cell->level() <  isosurface.min_refinement)
+                            {
+                              refine = true;
+                              break;
+                            }
+
+                          // If the current refinement level is larger than or equal to the maximum refinement
+                          // level, any refinement flag should be cleared.
+                          if (cell->level() >= isosurface.max_refinement)
+                            {
+                              clear_refine = true;
+                            }
+
+                          // If the current refinement level is larger than the maximum refinemment level,
+                          // a coarsening flag should be placed.
+                          if (cell->level() >  isosurface.max_refinement)
+                            {
+                              coarsen = true;
+                            }
+                        }
+                    }
+                }
+
+              // if both coarsen and refine are true, give preference to refinement
+              if (coarsen == true && refine == true)
+                coarsen = false;
+              if (refine == true)
+                clear_refine = false;
+              if (clear_coarsen == true)
+                coarsen = false;
+
+              // Perform the actual placement of the coarsening and refinement flags.
+              if (clear_coarsen == true)
+                {
+                  cell->clear_coarsen_flag ();
+                }
+              if (clear_refine == true)
+                {
+                  cell->clear_refine_flag ();
+                }
+              if (coarsen == true)
+                {
+                  cell->set_coarsen_flag ();
+                }
+              if (refine == true)
+                {
+                  cell->set_refine_flag ();
+                }
+            }
+        }
+    }
+
+    template <int dim>
+    void
+    Isosurfaces<dim>::
+    declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Mesh refinement");
+      {
+        prm.enter_subsection("Isosurfaces");
+        {
+          prm.declare_entry ("Isosurfaces", "depth",
```

Comment:
"depth" does not seem to be  sensible default. Maybe leave it empty?

Reply:
good point, I will just leave it empty.